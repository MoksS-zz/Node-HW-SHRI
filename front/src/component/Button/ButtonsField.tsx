import React from "react";
import { withNaming } from '@bem-react/classname';

const cn = withNaming({ e: '_', m: '__', v: '-' })

const styles = cn('ButtonsField');

const ButtonsField = (props) => {
  let classes;

  if (props.style === undefined) {
    classes = "ButtonsField"
  } else {
    classes = styles(props.style);
  }

  return (
    <div className={classes}>
      {props.children}
    </div>
  )

};

export default ButtonsField;