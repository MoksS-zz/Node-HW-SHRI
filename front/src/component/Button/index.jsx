import React from "react";
import { withNaming } from '@bem-react/classname';
import "./Buttons.scss";

const cn = withNaming({ e: '_', m: '__', v: '-' })

const styles = cn('Button');

const Button = (props) => {
  let classes;

  if (props.style === undefined) {
    classes = "Button Button__color-control Button__padding-control Button__height-default"
  } else {
    classes = styles(props.style);
  }

  return (
    <button 
      className={classes}
      onClick={props.onClick}>
      {props.children}
    </button>
  )

};

export default Button;