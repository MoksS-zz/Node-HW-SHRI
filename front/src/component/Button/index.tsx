import React, { MouseEvent, FC } from "react";
import cn from "../../helpers/bemCn";
import "./Buttons.scss";

const styles = cn('Button');

export interface ButtonProps {
  style?: {
    color?: "close" | "action" | "control";
    padding?: "action" | "control";
    indentRigth?: "s";
    indentBottom?: "s" | "xl";
    height?: "action" | "default";
    indentLeft?: "s"
  };
  onClick?(event: MouseEvent<HTMLButtonElement>): void;
}

const Button: FC<ButtonProps> = (props) => {
  let classes: string;

  if (props.style === undefined) {
    classes = "Button Button__color-control Button__padding-control Button__height-default"
  } else {
    classes = styles(props.style);
  }

  return (
    <button 
      className={classes}
      onClick={props.onClick}>
      {props.children}
    </button>
  )

};

export default Button;