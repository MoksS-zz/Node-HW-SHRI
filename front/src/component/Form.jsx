import React from "react";


const Form = ({header, descripton, children, name, formRef, method}) => {


  return (
    <form className="Form" name={name} ref={formRef} method={method} onSubmit={(e) => e.preventDefault()} onClick={(e) => e.preventDefault()}>
      <h2 className="Form_H2">{header}</h2>
      <p className="Form_Description">{descripton}</p>
      {children}
    </form>
  )
}

export default Form;