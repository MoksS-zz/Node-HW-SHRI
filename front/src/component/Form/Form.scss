.Form {

  width: 474px;

  &_H2 {
    font-style: normal;
    font-weight: 700;
    font-size: var(--font-size-xl);
    line-height: var(--line-height-l);
    margin: var(--space-xxxs) 0px;
    color: var(--black);
  }

  &_Description {
    font-weight: 400;
    font-size: var(--font-size-l);
    line-height: var(--line-height-m);
    letter-spacing: 0.06px;
    color: var(--grey-1000);
    margin: 0px 0px var(--space-xxl) 0px;
  }
}

@media only screen and (max-width: 824px) {
  .Form {
    width: 100%;

    &_Description {
      margin-bottom: 14px;
    }
  }
}