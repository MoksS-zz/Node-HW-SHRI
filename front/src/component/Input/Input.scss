.Input {
  margin: 0px 0px var(--space-l) 0px;

  &__flex-active {
    padding-top: 1px;
    display: flex;
    flex-direction: row;
    align-items: center;
  }
  &_Span {
    font-size:  var(--font-size-l);
    line-height: var(--line-height-m);
    margin: 0px 0px 0px var(--space-xxxs);
    letter-spacing: 0.06px;
    color: var(--black);
  }

  &_Head {
    margin: 0px var(--space-xxxs) var(--space-xxxxs) 0px;
    font-size:  var(--font-size-l);
    line-height: var(--line-height-m);
    letter-spacing: 0.06px;

    color: var(--black);

    &__indent {

      &-none {
        margin:0;
      }

      &-rigth {
        margin-right: var(--space-xxxs);
      }
    }
  }

  &_Field {
    height: 36px;
    padding: 9px var(--space-xxxs) var(--space-xxxs) 11px;
    display: flex;
    border: 2px solid var(--grey-400);
    justify-content: right;
    box-sizing: border-box;
    border-radius: var(--border-radius);

    &__position {
      &-rigth {
        padding: 9px 0px var(--space-xxxs) var(--space-xxl);
      }
    }
    
    &__sizeWidth {
      &-small {
        width: 52px;
      }
    }
  }

  &_Field:focus-within {
    border: 2px solid var(--grey-300);
  }

  &_Bind {
    color: red;
  }

  &_Input {
    padding: 0;
    flex-grow: 1;
    border: none;
    font-size:  var(--font-size-l);
    line-height: var(--line-height-xs);

    &__sizeWidth {
      &-small {
        width: 26px;
      }
    }
  }
  
  &_Input + .Button {
    display: none;
  }

  &_Input:not(:placeholder-shown) + .Button {
    display: block;
  }
  

  &_Input:focus, &_Input:active, &_Input:hover {
    border: none;
    outline: 0;
    outline-offset: 0;
  }
}