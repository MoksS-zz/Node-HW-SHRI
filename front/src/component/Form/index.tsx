import React, { FC, RefObject } from "react";
import "./Form.scss";

export interface FormProps {
  header: string;
  descripton: string;
  name: string;
  method: string;
  formRef: RefObject<HTMLFormElement>;
}

const Form: FC<FormProps> = ({header, descripton, children, name, formRef, method}) => {

  return (
    <form className="Form" name={name} ref={formRef} method={method} onSubmit={(e) => e.preventDefault()} onClick={(e) => e.preventDefault()}>
      <h2 className="Form_H2">{header}</h2>
      <p className="Form_Description">{descripton}</p>
      {children}
    </form>
  )
}

export default Form;