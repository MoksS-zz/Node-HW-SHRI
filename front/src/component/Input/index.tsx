import React, { useRef, FC } from "react";
import Button from "./../Button";
import Icon from "./../Icon";
import MaskedInput from 'react-text-mask'
import "./Input.scss";

export interface InputProps {
  bind?: boolean;
  head?: string;
  placeholder?: string;
  id?: string;
  textMask?: RegExp[];
  name?: string;
  value?: string | number;
  type?: string;
  span?: string; 
  onChange? (evt :React.ChangeEvent<HTMLInputElement>): void;
}

const Input: FC<InputProps> = (props) => {
  const inputEl = useRef<HTMLInputElement>(null);

  const clearInput = () => {
    if (inputEl.current) {
      inputEl.current.value = "";
    }
  }

  const bindInput = props.bind ? <span className="Input_Bind"> *</span> : "";
  let input;

  if (props.textMask !== undefined && Array.isArray(props.textMask)) {
    input = (
      <div className="Input Input__flex-active">
        <p className="Input_Head Input_Head__indent-none Input_Head__indent-rigth">{props.head}</p>
        <div className="Input_Field  Input_Field__sizeWidth-small Input_Field__position-rigth">
          <MaskedInput 
            mask={props.textMask} 
            className="Input_Input Input_Input__sizeWidth-small" 
            placeholder={props.placeholder}
            id={props.id} 
            name={props.name}
            defaultValue={props.value}
            placeholderChar={"\u2000"}
            onChange={props.onChange}
          />
          {/* <input className="Input_Input Input_Input__sizeWidth-small" type="text" maxLength="2" placeholder="10" name="minutes" defaultValue="10" /> */}
        </div>
        {props.span !== undefined ? <span className="Input_Span">{props.span}</span> : ""}
      </div>
    )
  } else {
    input = (
      <div className="Input">
        <p className="Input_Head">{props.head || ""}{bindInput}</p>
        <div className="Input_Field">
          <input
            className="Input_Input"
            type={props.type}
            placeholder={props.placeholder}
            name={props.name} 
            id={props.id}
            defaultValue={props.value}
            ref={inputEl}
          />
          <Button style={{ color: "close" }} onClick={clearInput}>
            <Icon style={{ size: "xs", img: "close" }} />
          </Button>
        </div>
        {props.span !== undefined ? <span className="Input_Span">{props.span}</span> : ""}
      </div>
    )
  }

  return input;
};

export default Input;