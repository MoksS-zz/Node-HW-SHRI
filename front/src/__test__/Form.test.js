import React, { useRef, useState } from "react";
import renderer from "react-test-renderer";
import Form from "../component/Form";
import Input from "../component/Input";
import Button from "../component/Button";
import ButtonsField from "../component/Button/ButtonsField";
import Text from "../component/Text";

jest.mock('react-text-mask', () => props => <input type="text" {...{ ...props }} />);
// здесь у меня таже проблема что и у этих реюят https://github.com/text-mask/text-mask/issues/427

const Stub = () => {
  const formRef = useRef(null);
  return (
    <Form
      header="Settings"
      descripton="Configure repository connection and synchronization settings."
      name="settings"
      method="post"
      formRef={formRef}
    >
      <Input
        head="GitHub repository"
        type="text"
        placeholder="user-name/repo-name"
        name="repoName"
        bind={true}
      />
      <Input
        head="Build command"
        type="text"
        placeholder="npm ci && npm run build"
        name="buildCommand"
        bind={true}
      />
      <Input
        head="Main branch"
        type="text"
        placeholder="master"
        name="mainBranch"
      />
      <Input
        head="Synchronize every"
        placeholder="10"
        name="period"
        span="minutes"
        textMask={[/\d/, /\d/]}
        value={123}
      />
      <Text style={{ size: "l", weight: "normal", lineHeight: "xxl", color: "red" }}>{""}</Text>
      <ButtonsField style={{ indent: "topXl" }}>
        <Button style={{ indentRigth: "s", indentBottom: "s", height: "action", color: "action", padding: "action" }}>
          <Text style={{ size: "m", lineHeight: "xxxxl", weight: "small", color: "default" }}>Save</Text>
        </Button>
        <Button style={{ color: "control", padding: "action", height: "action", indentBottom: "xl" }}>
          <Text style={{ size: "m", lineHeight: "xxxxl", weight: "small", color: "default" }}>Cancel</Text>
        </Button>
      </ButtonsField>
    </Form>
  )
}

it("renders Form correctly", () => {
  const form = renderer
    .create(<Stub/>)
    .toJSON();

  expect(form).toMatchSnapshot();
});
