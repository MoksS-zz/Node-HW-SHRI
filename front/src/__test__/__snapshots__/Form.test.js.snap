// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Form correctly 1`] = `
<form
  className="Form"
  method="post"
  name="settings"
  onClick={[Function]}
  onSubmit={[Function]}
>
  <h2
    className="Form_H2"
  >
    Settings
  </h2>
  <p
    className="Form_Description"
  >
    Configure repository connection and synchronization settings.
  </p>
  <div
    className="Input"
  >
    <p
      className="Input_Head"
    >
      GitHub repository
      <span
        className="Input_Bind"
      >
         *
      </span>
    </p>
    <div
      className="Input_Field"
    >
      <input
        className="Input_Input"
        name="repoName"
        placeholder="user-name/repo-name"
        type="text"
      />
      <button
        className="Button Button__color-close"
        onClick={[Function]}
      >
        <span
          className="Icon Icon__size-xs Icon__img-close"
        />
      </button>
    </div>
    
  </div>
  <div
    className="Input"
  >
    <p
      className="Input_Head"
    >
      Build command
      <span
        className="Input_Bind"
      >
         *
      </span>
    </p>
    <div
      className="Input_Field"
    >
      <input
        className="Input_Input"
        name="buildCommand"
        placeholder="npm ci && npm run build"
        type="text"
      />
      <button
        className="Button Button__color-close"
        onClick={[Function]}
      >
        <span
          className="Icon Icon__size-xs Icon__img-close"
        />
      </button>
    </div>
    
  </div>
  <div
    className="Input"
  >
    <p
      className="Input_Head"
    >
      Main branch
      
    </p>
    <div
      className="Input_Field"
    >
      <input
        className="Input_Input"
        name="mainBranch"
        placeholder="master"
        type="text"
      />
      <button
        className="Button Button__color-close"
        onClick={[Function]}
      >
        <span
          className="Icon Icon__size-xs Icon__img-close"
        />
      </button>
    </div>
    
  </div>
  <div
    className="Input Input__flex-active"
  >
    <p
      className="Input_Head Input_Head__indent-none Input_Head__indent-rigth"
    >
      Synchronize every
    </p>
    <div
      className="Input_Field  Input_Field__sizeWidth-small Input_Field__position-rigth"
    >
      <input
        className="Input_Input Input_Input__sizeWidth-small"
        defaultValue={123}
        mask={
          Array [
            /\\\\d/,
            /\\\\d/,
          ]
        }
        name="period"
        placeholder="10"
        placeholderChar="â€€"
        type="text"
      />
    </div>
    <span
      className="Input_Span"
    >
      minutes
    </span>
  </div>
  <span
    className="Text Text__size-l Text__weight-normal Text__lineHeight-xxl Text__color-red"
  >
    
  </span>
  <div
    className="ButtonsField ButtonsField__indent-topXl"
  >
    <button
      className="Button Button__indentRigth-s Button__indentBottom-s Button__height-action Button__color-action Button__padding-action"
    >
      <span
        className="Text Text__size-m Text__lineHeight-xxxxl Text__weight-small Text__color-default"
      >
        Save
      </span>
    </button>
    <button
      className="Button Button__color-control Button__padding-action Button__height-action Button__indentBottom-xl"
    >
      <span
        className="Text Text__size-m Text__lineHeight-xxxxl Text__weight-small Text__color-default"
      >
        Cancel
      </span>
    </button>
  </div>
</form>
`;
